### PR TITLE
feat: support custom pull request title

### LINKED ISSUE #783: [feature] support custom pull request title
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

 **Is your feature request related to a problem? Please describe.**

Add emjoy to commit message.

 **Describe the solution you'd like**

Support custom pull request title.

 **Describe alternatives you've considered**

Add a option for release-pr.

 **Additional context**

### PR DESCRIPTION
**Pull request by yi-Xu-0100**
add option `--pull-request-title-pattern` for the `cli`.
add option `pullRequestTitlePattern` for the `release-pr`.

The option adds title pattern to make release PR title, defaults to using `chore${scope}: release${component} ${version}`.

and example custom `pullRequestTitlePattern` is: `chore${scope}: 🔖 release${component} ${version}`.

The pattern need three parts which are `${scope}`, `${component}`, and `${version}`, and will throw an Error when missing one of them.

Fixes #783 🦕

### CODE REVIEW

**Comment by bcoe** [comment]
@yi-Xu-0100 thank you for the patch, will work on getting you review soon 👍

**Comment by yi-Xu-0100** [comment]
I can't wait to use this feature, so any reviews here?

@bcoe If this PR merged, I will make a PR for the release-please-action soon. Thanks for your reviews! 😀

File: src/util/pull-request-title.ts [hunk 1]
```diff
@@ -16,59 +16,106 @@
 // at the script level are undefined, they are only defined inside function
 // or instance methods/properties.
 
-const DEFAULT_PATTERN = /^chore(\((?<branch>[\w-.]+)\))?: release ?(?<component>[\w-.]*)? v?(?<version>[0-9].*)$/;
+const DEFAULT_PR_TITLE_PATTERN =
+  'chore${scope}: release${component} ${version}';
```
Comment (bcoe) [inline src/util/pull-request-title.ts:20]: I'm a little worried that this loses some of the subtleties of the regex, such as the fact that 
```suggestion
const DEFAULT_PR_TITLE_PATTERN = 'chore${scope}: release${component} ${version}';
```

**Review by bcoe — COMMENTED** [review]
This adds a bit of complexity, I'm curious what your use case is?

Happy to add this functionality if there's a good reason for it.

**Comment by bcoe** [comment]
@yi-Xu-0100 this work looks good to me, but I'd like to get @chingor13 to take a look, since he's the one who built out the title helper class for the Java release.

File: src/util/pull-request-title.ts (hunk 1 above)
Comment (yi-Xu-0100) [reply src/util/pull-request-title.ts:20]: I did not see the difference between this suggest commit to my commit. I have been using the `npm run fix` to format my commit.

**Comment by yi-Xu-0100** [comment]
@bcoe I keep the used regex generated with the `DEFAULT_PR_TITLE_PATTERN` as the same as regex used before by replacing string. And I use the feature for keeping my commit message in the same pattern as before, which will allow my do not modify the PR title to squash merging. 😀

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@yi-Xu-0100 talked with my peers yesterday, and we're happy with the change. We'd like to eventually remove checking for version in the issue title, at which point we can remove some of the logic you've written.

**Comment by yi-Xu-0100** [comment]
@bcoe I'm not sure what I should do with it. You mean the code of `error of missing version` can be deleted? Should I make a new PR?

https://github.com/googleapis/release-please/blob/0c09488da5049172ed69a68b014d5a78da374e18/src/util/pull-request-title.ts#L32-L36

**Pull request merged**